scrapy/downloadermiddlewares/decompression.py

```diff
@@ -30,10 +30,11 @@ def __init__(self):
 
     def _is_tar(self, response):
         archive = BytesIO(response.body)
-        try:
-            tar_file = tarfile.open(name=mktemp(), fileobj=archive)
-        except tarfile.ReadError:
-            return
+        with NamedTemporaryFile() as temporary_file:
+            try:
+                tar_file = tarfile.open(name=temporary_file.name, fileobj=archive)
```

Comment:
[According to the documentation](https://docs.python.org/3/library/tarfile.html#tarfile.TarFile), the `name` parameter is optional. [We added it 13 years ago](https://github.com/scrapy/scrapy/commit/9022f84c05a13d58a03f24cc65439ccc109ee57b) because “None is failing in some cases”. But I have no idea what those cases would be. It may be that the Python version back then had a bug or something.

I am thinking that we could remove the parameter altogether. If anyone can provide a scenario where it fails, we can try to address that scenario in a point release of Scrapy, this time documenting the issue and adding a test for it, which would be best in the long run.

Reply:
This became irrelevant with the removal of the entire module.